compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -93,7 +93,9 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
         checkAnnotations(tree)
         healInfo(tree, tree.srcPos)
         super.transform(tree)
-
+      case tree: TypeDef if tree.symbol.is(Case) && level > 0 =>
+        report.error(reporting.CaseClassInInlinedCode(tree), tree)
+        super.transform(tree)
```

Comment:
Case classes are just desugarings. Maybe generalize it to all `TypeDef`?

Reply:
What do you mean by generalizing? Do you mean not allowing these ones?
```
'{ class Bar extends Foo { def foo = ... }; ... }
'{ type T = ...;  }
```
Those are necessary.

Reply:
If we don't generalize, the rule seems arbitrary and will not do the job. If a programmer just hand-crafted the desugaring the a case class, doesn't it have the same effect as a case class?